server/models/documents.coffee

```diff
@@ -21,9 +21,17 @@ RemoteStorageDoc.asStore =
         console.log "SET CALLED", u, key, buf
         RemoteStorageDoc.byKey key, (err, doc) ->
             return cb err if err
+
+            # SET undefined === DELETE
+            if not buf?
+                if not doc?
+                    console.error 'Document to delete not found!'
+                    return cb 'Document not found'

```

Comment:
At Cozy we decided to not have any return in the middle of the function. We keep them only for `callback err` cases. Can you change it with indentation?